### DISCUSSION TITLE
Cannot install working formula

### DISCUSSION THREAD

**Discussion by Ninzero** [Homebrew/discussions#4221] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.21-166-g6f195c1
ORIGIN: https://github.com/Homebrew/brew
HEAD: 6f195c1e6fd29a6574dbe94428ae4c3324863aa5
Last commit: 16 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 44f017074ab5fdccc07a6c8a6055839c1de2d000
Core tap last commit: 14 hours ago
Core tap branch: add-leetup
Core tap JSON: 16 Feb 18:37 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_AUTO_UPDATE_SECS: 600
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: code -w
HOMEBREW_MAKE_JOBS: 12
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_INSTALL_CLEANUP: set
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit arm_blizzard_avalanche
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2.1-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: The filesystem on /System/Volumes/Data appears to be case-sensitive.
The default macOS filesystem is case-insensitive. Please report any apparent problems.

Warning: You have uncommitted modifications to Homebrew/homebrew-core.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /opt/homebrew/Library/Taps/homebrew/homebrew-core && git stash -u && git clean -d -f

Uncommitted files:
   M Formula/leetup.rb

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/core) checkout master
```

### Description of issue

I am working on adding a new formula called leetup. According to [Formula Cookbook](https://docs.brew.sh/Formula-Cookbook#install-the-formula), I should try to install this new formula with `brew install --build-from-source --verbose --debug leetup`. However, this error occurs:

> ==> Fetching leetup
curl: (22) The requested URL returned error: 404
>
>Error: Failure while executing; `/usr/bin/env /opt/homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/3.6.21-166-g6f195c1\ \(Macintosh\;\ arm64\ Mac\ OS\ X\ 13.2.1\)\ curl/7.86.0 --header Accept-Language:\ en --fail --progress-bar --retry 3 --location --remote-time --output /Users/Ninzero/Library/Caches/Homebrew/Formula/leetup.rb https://raw.githubusercontent.com/Homebrew/homebrew-core/44f017074ab5fdccc07a6c8a6055839c1de2d000/Formula/leetup.rb` exited with 22. Here's the output:
curl: (22) The requested URL returned error: 404

It feels like homebrew is trying to get `leetup.rb` from homebrew-core repo, which does not exist since I am still working on this new formula.

My current finished part of `leetup.rb`:

``` Ruby
class Leetup < Formula
  desc "Command line tool to solve Leetcode problems."
  homepage "https://github.com/dragfire/leetup"
  url "https://github.com/dragfire/leetup/archive/v1.1.0.tar.gz"
  sha256 "33e4be4278c72d09a8f20615aa4dbf272ea3087750c3ba31bdaadada4cf57bc1"
  license "MIT"
  head "https://github.com/dragfire/leetup.git", branch: "master"

  depends_on "rust" => :build

  def install
    system "cargo", "install", *std_cargo_args
  end

  test do
    # I haven't added a test.
    system "false"
  end
end
```

**Comment by ZhongRuoyu** [comment, marked as answer]
Please try `export HOMEBREW_NO_INSTALL_FROM_API=1` before installing.

**Reply by Ninzero** [reply]
That works. Thank you for the help! Please consider adding this line to the formula cookbook since it is necessary for testing new formulae.

**Comment by ekpotosammuel** [comment]
I am having an issue installing Fomular as well here is my error message 

==> Fetching aom
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/1ddfb0b1a52079dc2c8f0ee295302b7ff14a813e/Formula/a/aom.rb
Already downloaded: /Users/ekpoto/Library/Caches/Homebrew/downloads/1e2b6f663c7964146e3e33b309a9868e35e77610d4065a11a5249183c5457d78--aom.rb
Error: aom: Failed to download resource "aom--homebrew-bus_qcif_15fps.y4m"
Failure while executing; `/usr/bin/env /usr/local/Homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null --globoff --show-error --user-agent Homebrew/4.1.12-2-g1f80e82\ \(Macintosh\;\ Intel\ Mac\ OS\ X\ 10.15.7\)\ curl/7.64.1 --header Accept-Language:\ en --retry 3 --fail --location --silent --head https://media.xiph.org/video/derf/y4m/bus_qcif_15fps.y4m` exited with 7. Here's the output:
curl: (7) Failed to connect to media.xiph.org port 443: Connection refused

**Reply by ZhongRuoyu** [reply]
This is a problem with your connection to https://media.xiph.org/video/derf/y4m/bus_qcif_15fps.y4m. If this error persists even if you retry, you'll need to figure out why you can't connect.
